models/project/project.go

```diff
@@ -78,12 +81,15 @@ func (err ErrProjectBoardNotExist) Unwrap() error {
 
 // Project represents a project board
 type Project struct {
-	ID          int64  `xorm:"pk autoincr"`
-	Title       string `xorm:"INDEX NOT NULL"`
-	Description string `xorm:"TEXT"`
-	RepoID      int64  `xorm:"INDEX"`
-	CreatorID   int64  `xorm:"NOT NULL"`
-	IsClosed    bool   `xorm:"INDEX"`
+	ID          int64                  `xorm:"pk autoincr"`
+	Title       string                 `xorm:"INDEX NOT NULL"`
+	Description string                 `xorm:"TEXT"`
+	OwnerID     int64                  `xorm:"INDEX"`
+	Owner       *user_model.User       `xorm:"-"`
+	RepoID      int64                  `xorm:"INDEX"`
```

Comment:
```suggestion
	Title       string                 `xorm:"INDEX UNIQUE(title_owner_repo) NOT NULL"`
	Description string                 `xorm:"TEXT"`
	OwnerID     int64                  `xorm:"INDEX UNIQUE(title_owner_repo)"`
	Owner       *user_model.User       `xorm:"-"`
	RepoID      int64                  `xorm:"INDEX UNIQUE(title_owner_repo)"`
```
I think this constraint makes sense.
However, if we add it, we should probably mark this PR as breaking as it would require to rename existing duplicated titles (that should not have ever been used already)